app/controllers/admin/applications_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+  end
+
+  def update
+    @pet_application = PetApplication.find(params[:id])
+    @application = @pet_application.application
+    @pet = @pet_application.pet
```

Comment:
It's okay to make these instance vars here, but keep in mind that you can call `@pet_application.application` and `@pet_application.pet` from the view too! You're not violating MVC by doing that